src/libraries/System.Private.CoreLib/src/System/WeakReference.cs

```diff
@@ -2,11 +2,30 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using System.Runtime.Serialization;
 
+using static System.WeakReferenceHandleTags;
+
 namespace System
 {
+    internal static class WeakReferenceHandleTags
+    {
+        // the lowermost bit is used to indicate whether the handle is tracking resurrection
+        // handles are at least 16bit aligned, so we can use one bit for tagging
```

Comment:
Nit; I think alignment is usually measured in bytes, not bits.

Reply:
Will switch to bytes. I heard 64-bit/32-bit aligned, but 16-bit sounds odd.